trento/migration/sso-integration.md

```diff
@@ -0,0 +1,390 @@
+
+```bash
+docker run -d \
+-p 4000:4000 \
+--name trento-web \
+--network trento-net \
+--add-host "host.docker.internal:host-gateway" \
+
+...[other settings]...
+
+-e ENABLE_OAUTH2=true  \
+-e OAUTH2_CLIENT_ID=<<OAUTH2_CLIENT_ID>> \
+-e OAUTH2_CLIENT_SECRET=<<OAUTH2_CLIENT_SECRET>> \
+-e OAUTH2_BASE_URL=<<OAUTH2_BASE_URL>> \
+-e OAUTH2_AUTHORIZE_URL=<<OAUTH2_AUTHORIZE_URL>> \
+-e OAUTH2_TOKEN_URL=<<OAUTH2_TOKEN_URL>> \
+-e OAUTH2_USER_URL=<<OAUTH2_USER_URL>> \
+
+# Optional:
+-e OAUTH2_SCOPES=<<OAUTH2_SCOPES>>  \
+-e OAUTH2_CALLBACK_URL=<<OAUTH2_CALLBACK_URL>> \
+
+...[other settings]...
+```
+
+### Available variables
+
+OAUTH2_CLIENT_ID
+
+: OAUTH2 client id
+
+OAUTH2_CLIENT_SECRET
+
+: OAUTH2 client secret
+
+OAUTH2_BASE_URL
+
+: OAUTH2 base url
+
+OAUTH2_AUTHORIZE_URL
+
+: OAUTH2 authorization url
+
+OAUTH2_TOKEN_URL
+
+: OAUTH2 token url
+
+OAUTH2_USER_URL
+
+: OAUTH2 token url
+
+OAUTH2_SCOPES
+
+: OAUTH2 scopes, used to define the user values sent to the SP. It must be adjusted depending on IDP provider requirements (default value: `openid profile email`)
+
+OAUTH2_CALLBACK_URL
+
+: OAUTH2 callback url where the IDP is redirecting once the authentication is completed (default value: `https://#{TRENTO_WEB_ORIGIN}/auth/oauth2_callback`)
+
+## SAML
+
+Trento integrates with an IDP that uses the SAML protocol to authenticate users accessing the Trento web console.
+
+In order to use an existing SAML IDP, some requirements must be met, configuring the IDP and starting Trento in a specific way. Follow the next instructions to properly setup both.
+
+### SAML IDP setup
+
+Configure the existing IDP with the next minimum options to be able to connect with Trento's Service Provider (SP).
+
+#### SAML user profile
+
+Users provided by the SAML installation must have some few mandatory attributes in order to login in Trento. All of them are mandatory, even though their name is configurable.
+The user profile must include attributes for: username, email, first name and last name.
+This attributes must be mapped for all users wanting to connect to Trento.
+
+By default, Trento expects the `username`, `email`, `firstName` and `lastName` attribute names. All these 4 attribute names are configurable using the next environment variables, following the same order: `SAML_USERNAME_ATTR_NAME`, `SAML_EMAIL_ATTR_NAME`, `SAML_FIRSTNAME_ATTR_NAME` and `SAML_LASTNAME_ATTR_NAME`.
+So for example, if the IDP user profile username is defined as `attr:username`, `SAML_USERNAME_ATTR_NAME=attr:username` should be used.
+
+#### Signing keys
+
+Commonly, SAML protocol messages are signed with SSL. This is optional using Trento, and the signing is not required (even though it is recommended).
+If the IDP signs the messages, and expect signed messages back, certificates used by the SP (Trento in this case) must be provided to the IDP, the Certificate file in this case.
```

Comment:
What's "SP" here? Service Provider? I've splitted the sentence a bit. But I'm not sure how to use the "the certificate file in this case". Could you clarify that?

```suggestion
If the IDP signs the messages and expect signed messages back. Certificates used by the Trento service provider must be provided to the IDP, the certificate file in this case.
```

Reply:
Let me rephrase this and move it to the top chapter, as a background information